// Proto file representing the EDB IO Component Property class

syntax = "proto3";

// Package
package ansys.api.edb.v1;

// Imports
import "edb_messages.proto";

// IO Component Property service definition
service IOComponentPropertyService {
  // Create IO Component Property 
  rpc Create(google.protobuf.Empty) returns (EDBObjMessage) {}

  // Get/Set the solder ball property
  rpc SetSolderBallProperty(PointerPropertyMessage) returns (google.protobuf.Empty) {}
  rpc GetSolderBallProperty(EDBObjMessage) returns (EDBObjMessage) {}

  // Get/Set the port property
  rpc SetPortProperty(PointerPropertyMessage) returns (google.protobuf.Empty) {}
  rpc GetPortProperty(EDBObjMessage) returns (EDBObjMessage) {}
}